formValidation.js

```diff
@@ -0,0 +1,32 @@
+(function(){
+    const dataTypesRegExp = {
+        address: /^0x[0-9a-f]{40}$/i
+    };
+
+    window.formValidation = {
+        validateArgsByEvent: validateArgsByEvent,
+        dataTypesRegExp: dataTypesRegExp
+    };
+
+    function validateArgsByEvent(args, methodName) {
+        let resultPromise = Promise.resolve([]);
+
+        args.forEach((argNode) => {
+            resultPromise = resultPromise.then(validateArg.bind(this, argNode, methodName));
+        });
+        return resultPromise;
```

Comment:
Could we shorten this function like below?

```
function validateArgsByEvent(args, methodName) {
    return Promise.all(args.map(argNode => validateArg(argNode, methodName)));
}
```

Reply:
I believe, we can